frontend/lib/parsing/ParserContextImpl.h

```diff
@@ -1296,34 +1289,31 @@ AstNode* ParserContext::buildNewExpr(YYLTYPE location,
                                                  fnCall);
       child.reset(wrappedFn);
       return expr;
-    } else {
-      //something wrong, as below
-      this->raiseError(location, "Invalid form for 'new' expression");
```

Comment:
Yep, good catch. I didn't realize there were two separate instances of this message being raised and I had only kept one. I'll add a specialized error class for this real quick since it shows up twice, and I'd like to cover it with a test.